Community-Supported/hyper-jupyter-kernel/README.md

```diff
@@ -0,0 +1,141 @@
+
+It integrates with the execution framework provided by the [Jupyter project](https://jupyter.org/).
+Thereby, the `hyper_kernel` only has to take care of the actual execution of the SQL statements, but doesn't need to provide its own cross-platform text interface/UI, command-line history etc.
+
+Being a Jupyter kernel, the `hyper_kernel` can be used in combination with the rest of the Jupyter ecosystem.
+In particular:
+* It can be run from an old-school or over a ssh-connection using `jupyter console --kernel=hyper`
+* It can be used from JupyterLab's consoles, providing a more modern UI experience
+* It can be used with JupyterLab notebooks, allowing to tell stories based on your SQL queries
+
+## Get started using JupyterLab
+
+JupyterLab is a web-based IDE for data-scientist.
+With the `hyper_kernel` you can integrate HyperAPI's SQL interface directly into this IDE.
+
+![Screenshot of jupyter console](media/jupyterlab.png)
+
+First, install jupyter-console and the kernel provided in this repository using
+
+```
+python -m pip install jupyter-console tableauhyperapi git+https://github.com/tableau/hyper-api-samples#subdirectory=Community-Supported/hyper-jupyter-kernel
+python -m hyper_kernel.install
+```
+
+Next launch JupyterLab using
+
+```
+python -m jupyter lab
+```
+
+and open JupyterLab in your browser
+
+
+## Get started using the command-line interface
+
+![Screenshot of jupyter console](media/jupyter_console.png)
+
+First, install jupyter-console and the kernel provided in this repository using
+
+```
+python -m pip install jupyter-console tableauhyperapi git+https://github.com/tableau/hyper-api-samples#subdirectory=Community-Supported/hyper-jupyter-kernel
+python -m hyper_kernel.install
+```
+
+Afterwards, you can launch the console using
+
+```
+jupyter console --kernel=hyper
+```
+
+## Running your first query
+
+Let's start with something simple: Enter
+
+```
+SELECT 12/2 AS result
+```
+
+into your console and submit it. You should get `6` as a result. Congrats! You just executed your first query.
+
+# Quick start for common use cases
+
+Independent of how you run your `hyper_kernel`, the usage is always similar:
+You feed SQL to it, and in return you get the query results.
+The [HyperAPI SQL reference](https://help.tableau.com/current/api/hyper_api/en-us/reference/sql/index.html) contains all the information you need.
+For everyone who prefers to not read the whole reference first, here is a short quick start on the most common scenarios:
+
+
+## Creating a Hyper database
+
+That was boring, though... Instead of only using Hyper as your calculator, you usually want to use it to crunch some data.
+
+First let's [create a new database](https://help.tableau.com/current/api/hyper_api/en-us/reference/sql/sql-createdatabase.html):
+```
+CREATE DATABASE "my_data.hyper";
+```
+
+and load that newly created database into our session:
+```
+ATTACH DATABASE "my_data.hyper" AS database1;
```

Comment:
`ATTACH DATABASE` is not documented. We will need to wait until we document it officially. Additionally, we need to discuss whether we see a concern with it

Reply:
So what are you proposing instead?

Reply:
I'm not proposing to do sth differently. I want to make it clear that we first need to update the documentation, which means we need to wait for the November release. And that we need to discuss this with the team. I don't see a concern with it, but maybe others do

Reply:
found a way to workaround the issue which doesn't require us to wait for ATTACH to be documented.
I simply wrapped `catalog.attach_database` into client-side commands `\attach` and `\detach`